pulsar-broker/src/main/java/org/apache/pulsar/broker/service/schema/KeyValueSchemaCompatibilityCheck.java

```diff
@@ -0,0 +1,138 @@
+        ByteBuffer byteBuffer = ByteBuffer.wrap(bytes);
+        int keyLength = byteBuffer.getInt();
+        byte[] keySchema = new byte[keyLength];
+        byteBuffer.get(keySchema);
+
+        int valueLength = byteBuffer.getInt();
+        byte[] valueSchema = new byte[valueLength];
+        byteBuffer.get(valueSchema);
+        return new KeyValue<>(keySchema, valueSchema);
+    }
+
+    private SchemaType fetchSchemaType(Map<String, String> properties, String key) {
+        if (properties.get(key) != null) {
+            return SchemaType.valueOf(properties.get(key));
+        }
+        return SchemaType.BYTES;
+    }
+
+    private SchemaData fetchSchemaData(
+            byte[] keyValue, SchemaType schemaType, Gson schemaGson, Map<String, String> properties, String key) {
+        if (properties.get(key) != null) {
+            return SchemaData.builder().data(keyValue)
+                    .type(schemaType)
+                    .props(schemaGson.fromJson(properties.get(key), Map.class)).build();
+        }
+        return SchemaData.builder().data(keyValue)
+                .type(schemaType)
+                .props(Collections.emptyMap()).build();
+    }
+
+    private KeyValue<SchemaData, SchemaData> parseSchemaData(SchemaData schemaData) {
+        KeyValue<byte[], byte[]> keyValue = this.splitKeyValueSchemaData(schemaData.getData());
+        Map<String, String> properties = schemaData.getProps();
+        SchemaType keyType = fetchSchemaType(properties, "key.schema.type");
+        SchemaType valueType = fetchSchemaType(properties, "value.schema.type");
+        Gson schemaGson = new Gson();
+        SchemaData keySchemaData = fetchSchemaData(
+                keyValue.getKey(), keyType, schemaGson, properties, "key.schema.properties");
+        SchemaData valueSchemaData = fetchSchemaData(
+                keyValue.getValue(), valueType, schemaGson, properties, "value.schema.properties");
+        return new KeyValue<>(keySchemaData, valueSchemaData);
+    }
+
+    @Override
+    public SchemaType getSchemaType() {
+        return SchemaType.KEY_VALUE;
+    }
+
+    @Override
+    public boolean isWellFormed(SchemaData to) {
+        KeyValue<SchemaData, SchemaData> keyValue = parseSchemaData(to);
+        SchemaCompatibilityCheck valueCheck;
+        if (checkers.get(keyValue.getValue().getType()) != null) {
+            valueCheck = checkers.get(keyValue.getValue().getType());
+        } else {
+            valueCheck = SchemaCompatibilityCheck.DEFAULT;
+        }
+        return valueCheck.isWellFormed(keyValue.getKey()) && valueCheck.isWellFormed(keyValue.getValue());
+    }
+
+    @Override
+    public boolean isCompatible(SchemaData from, SchemaData to, SchemaCompatibilityStrategy strategy) {
+        if (from.getType() != SchemaType.KEY_VALUE || to.getType() != SchemaType.KEY_VALUE) {
+            if (strategy == SchemaCompatibilityStrategy.FULL || strategy == SchemaCompatibilityStrategy.ALWAYS_COMPATIBLE) {
```

Comment:
I don't understand why we are returning `true` when `strategy == SchemaCompatibilityStrategy.FULL`. Can you explain more?

Reply:
When I debug integration test of debezium, I found its default policy is FULL, schemaData of from is String and schemaData of to is KeyValue,  i am not sure whether this judgment condition should be added here or whether the strategy should be updated to ALWAYS_COMPATIBLE in the integration test:
```
10:03:34.332 [ForkJoinPool.commonPool-worker-2] INFO  org.apache.pulsar.broker.service.schema.KeyValueSchemaCompatibilityCheck - from SchemaData(type=STRING, isDeleted=false, timestamp=0, user=, data=[], props={})
10:03:34.332 [ForkJoinPool.commonPool-worker-2] INFO  org.apache.pulsar.broker.service.schema.KeyValueSchemaCompatibilityCheck - to SchemaData(type=KEY_VALUE, isDeleted=false, timestamp=1558087414170, user=, data=[0, 0, 0, 0, 0, 0, 0, 0], props={key.schema.properties={}, value.schema.properties={}, value.schema.type=BYTES, key.schema.name=Bytes, value.schema.name=Bytes, kv.encoding.type=INLINE, key.schema.type=BYTES})
```

Reply:
@tuteng I think you need to fix integration tests. 

1) a `FULL` policy means it has to be both backward and forward compatible with previous schema. So you can skip the validation when the policy is `FULL`.

2) The whole point of enabling key/value schema compatibility check is to enforce and allow schema evolutions. It doesn't make any senses to bypass it.

3) You need to figure out why the integration tests introduces `STRING` schema first and fix the integration test. You shouldn't mask the error in order to make a wrong integration test pass.